File: source/Core/Calculation/SinglePhase/SingleVoltageLevel/PQBus.cs
using System;
using System.Numerics;

namespace Calculation.SinglePhase.SingleVoltageLevel
{
    public class PQBus
    {
        private readonly int _id;
        private readonly Complex _power;

        public PQBus(int ID, Complex power)
        {

            if (ID < 0)
                throw new ArgumentOutOfRangeException("ID", "mustn't be negative");

            _id = ID;
            _power = power;
        }

        public int ID
        {
            get { return _id; }
        }

        public Complex Power
        {
            get { return _power; }
        }
    }
}

File: source/Core/HELM/PQBus.cpp
#include "PQBus.h"
#include <assert.h>

using namespace std;

PQBus::PQBus() :
	_id(-1),
	_power(0, 0)
{ }

PQBus::PQBus(int id, Complex<long double> power) :
	_id(id),
	_power(power)
{
	assert(_id >= 0);
}

int PQBus::getId() const
{
	return _id;
}

const Complex<long double>& PQBus::getPower() const
{
	return _power;
}
